arch/x86/kvm/pkvm/vmx/ept.c

```diff
@@ -264,3 +265,33 @@ int pkvm_handle_host_ept_violation(void)
 	pkvm_host_mmu_unlock();
 	return ret;
 }
+
+void pkvm_host_ept_finalize(void)
+{
+	struct kvm_vcpu *hvcpu = this_cpu_read(host_vcpu);
+
+	if (!host_ept)
+		return;
+
+	secondary_exec_controls_setbit(to_vmx(hvcpu), SECONDARY_EXEC_ENABLE_EPT);
+	vmcs_write64(EPT_POINTER, host_ept->root);
+
+	/* enable vpid */
+	if ((host_vmcs_config.cpu_based_2nd_exec_ctrl & SECONDARY_EXEC_ENABLE_VPID) &&
+	    (cpu_has_vmx_invvpid_single() || cpu_has_vmx_invvpid_global()) &&
+	    cpu_has_vmx_invvpid()) {
```

Comment:
nit: how about adding yet another check to this `if`: `pkvm_host_vpid > 1`? So that even on a hypothetical machine with 65536 CPUs, if we run out of VPIDs, we are still safe? :D